pkg/parser/boolevator/boolevator.go

```diff
@@ -45,9 +45,15 @@ func (boolevator *Boolevator) Eval(expr string, env map[string]string) (bool, er
 	// Work around text/scanner stopping at newline when scanning strings
 	expr = strings.ReplaceAll(expr, "\n", "\\n")
 
-	// Pre-expand environment variables
+	// Ensure that we keep the env as is
+	localEnv := make(map[string]string)
 	for key, value := range env {
-		env[key] = expander.ExpandEnvironmentVariables(value, env)
+		localEnv[key] = value
+	}
+
+	// Pre-expand environment variables
+	for key, value := range localEnv {
+		localEnv[key] = expander.ExpandEnvironmentVariables(value, localEnv)
```

Comment:
Can it be replaced with?

```suggestion
		localEnv[key] = expander.ExpandEnvironmentVariables(value, env)
```